src/services/PromotionService.ts

```diff
@@ -0,0 +1,113 @@
+import { FilterOptions, Promotion, Sort } from '../types/promotion';
+import Routes from '../utils/routes';
+
+type PromotionsSuccess = Promotion[];
+
+type PromotionsError = {
+  errorCode: string;
+  message: string[];
+};
+
+type PromotionsResponse = PromotionsSuccess | PromotionsError;
+
+/**
+ * Fetches entire list of promotions. If a query object is given, filters the promotions according to the given query.
+ * If an error occurs, an empty list will be returned.
+ *
+ * @param query [optional] - An array of objects with key-value pairs for the query parameters
+ */
+export async function getPromotions(query?: Record<string, string>[]): Promise<Promotion[]> {
```

Comment:
Thanks for the documentation, this helps a lot

Reply:
Just out of curiosity, what's the reason behind using functions rather than classes?

Reply:
@ericrkuo Good question, I think initially I tried using classes but they were hard to manage. Either I would have to instantiate it then pass that instance down the component hierarchy or I could make it a singleton instance (and not sure if I liked that). But this was before I added state management so now it might be easier to do since I can instantiate a Promotions object and then store that instance in the state object. I think I'll actually explore this idea, thanks for bringing it up :)

Reply:
Ohhh I see, that makes sense!